### PR TITLE
Direct listing links only show listing if published

### PR DESCRIPTION
**Pull request by mariocsee**
## What type of PR is this? (check all applicable) 
- [x] Bug Fix

## Description
Direct links of listings show even expired or unpublished listings. This changes `@displayed_classified_listing` to find the slug from only published listings. 

## Related Tickets & Documents
N/A

## Mobile & Desktop Screenshots/Recordings (if there are UI changes)
N/A

## Added to documentation?

- [x] no documentation needed

### CODE REVIEW

**Comment by maestromac** [comment]
So what happens now if direct link of an expired listing was accessed?

File: app/controllers/classified_listings_controller.rb [hunk 1]
```diff
@@ -6,14 +6,15 @@ class ClassifiedListingsController < ApplicationController
   before_action :authenticate_user!, only: %i[edit update new dashboard]
 
   def index
-    @displayed_classified_listing = ClassifiedListing.find_by!(slug: params[:slug]) if params[:slug]
+    published_listings = ClassifiedListing.where(published: true)
+    @displayed_classified_listing = published_listings.find_by(slug: params[:slug]) if params[:slug]
```
Comment (rhymes) [inline app/controllers/classified_listings_controller.rb]: Should this still be `find_by!` with the exclamation point?

Comment (mariocsee) [reply app/controllers/classified_listings_controller.rb]: Oops! Let me bring it back. Thanks, @rhymes!

**Comment by mariocsee** [comment]
> So what happens now if direct link of an expired listing was accessed?

Direct links to listings that are unpublished (expired and WIP, drafts) lead to `/listings/:category` so it will show other active listings in said category but will not display the expired listing.

**Comment by rhymes** [comment]
You might want to add a test to make sure that only published listing can be fetched

File: spec/requests/classified_listings_spec.rb [hunk 2]
```diff
@@ -14,6 +14,45 @@
...
+        expect(response.body).to include(listing.title)
+      end
+    end
+
+    context "when the user has category and slug params for expired listing" do
+      it "shows only active listings from that category" do
+        expired_listing.published = false
+        expired_listing.save
+        get "/listings", params: { category: `#{expired_listing.category}`, slug: `#{expired_listing.slug}` }
+        expect(response.body).not_to include(expired_listing.title)
```
Comment (rhymes) [inline spec/requests/classified_listings_spec.rb]: this is not a 404, correct?

Comment (mariocsee) [reply spec/requests/classified_listings_spec.rb]: Nope! It just doesn't pass the expired listing to Preact and therefore just renders a page with listings of that category.

**Pull request merged**